pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java

```diff
@@ -413,10 +413,21 @@ protected void configure() {
       }
     });
 
-    _adminApp.start(_listenerConfigs);
+    TlsConfig tlsDefaults = TlsUtils.extractTlsConfig(_config, ControllerConf.CONTROLLER_TLS_PREFIX);
+
+    // install default SSL context if necessary
+    if (CommonConstants.HTTPS_PROTOCOL.equals(_config.getProperty(ControllerConf.CONTROLLER_BROKER_PROTOCOL))) {
+      LOGGER.info("Installing default SSL context for broker relay requests");
+      TlsConfig tlsConfig = TlsUtils.extractTlsConfig(tlsDefaults, _config, ControllerConf.CONTROLLER_BROKER_TLS_PREFIX);
```

Comment:
Does it make sense to move this to ListenerConfigUtil.buildControllerConfigs() method? It has all the data needed to install the right factory. Correct?

Reply:
imo "principle of least surprise" applies here. If I create a config pojo I wouldn't expect the side-effect of someone rewriting my ssl-factories